app/controllers/reviews_controller.rb

```diff
@@ -19,6 +19,16 @@ def create
     end
   end
 
+  def edit
+    @review = Review.find(params[:id])
+
+    if !user_signed_in?
+      redirect_to monster_path(@review.monster_id), notice: "Please sign in to edit reviews"
+    elsif @review.user_id != current_user.id
+      redirect_to monster_path(@review.monster_id), notice: "You are not the author of this review"
+    end
```

Comment:
Good attention to detail here.  I'd encourage you to also look up using the controller to restrict access to actions via things like `before_action :authenticate_user!, :except => [:show, :index]`